nsv13/code/game/gamemodes/pvp/pvp.dm

```diff
@@ -0,0 +1,237 @@
+	var/obj/structure/overmap/syndiship
+	for(var/z in SSmapping.levels_by_trait(ZTRAIT_CORVI))
+		syndiship = new ship_type(get_turf(locate(round(world.maxx * 0.5, 1), round(world.maxy * 0.5, 1), z))) //Make a new syndie ship object in Corvi.
+	message_admins("Spawning in syndi ship map, this may take a while. No the game hasn't crashed, I'm just loading a map before we start.") //Warn the admins. This shit takes a while.
+	var/list/errorList = list()
+	var/list/loaded = SSmapping.LoadGroup(errorList, "Syndicate ship", "map_files/PVP", map_file, default_traits = ZTRAITS_MINING_SHIP, silent = FALSE)
+	if(errorList.len)	// reebe failed to load
+		message_admins("Syndie ship failed to load!")
+		log_game("Syndie ship failed to load!")
+		return FALSE
+	for(var/datum/parsed_map/PM in loaded)
+		PM.initTemplateBounds()
+	repopulate_sorted_areas()
+	var/n_agents = antag_candidates.len
+	if(n_agents >= enemies_to_spawn)
+		for(var/i = 0, i < enemies_to_spawn, i++)
+			var/datum/mind/new_op = pick_n_take(antag_candidates)
+			pre_nukeops += new_op
+			new_op.assigned_role = "Syndicate crewmember"
+			new_op.special_role = "Syndicate crewmember"
+			log_game("[key_name(new_op)] has been selected as a syndicate crewmember")
+		addtimer(CALLBACK(src, .proc/force_lighting, syndiship), 6 SECONDS)
+		return TRUE
+	else
+		qdel(syndiship)
+		setup_error = "Not enough syndicate crew candidates"
+		return FALSE
+////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////
+/datum/game_mode/pvp/proc/force_lighting(obj/structure/overmap/hammurabi)
+	for(var/area/AR in hammurabi.linked_areas) //Fucking force a lighting update IDEK why we have to do this but it just works
+		AR.set_dynamic_lighting(DYNAMIC_LIGHTING_DISABLED)
+		sleep(1)
+		AR.set_dynamic_lighting(DYNAMIC_LIGHTING_ENABLED)
+
+/datum/game_mode/pvp/post_setup()
+	//Assign leader
+	var/datum/mind/leader_mind = pre_nukeops[1]
+	var/datum/antagonist/nukeop/L = leader_mind.add_antag_datum(leader_antag_datum_type)
+	nuke_team = L.nuke_team
+	//Assign the remaining operatives
+	for(var/I in 2 to pre_nukeops.len)
+		var/datum/mind/nuke_mind = pre_nukeops[I]
+		nuke_mind?.add_antag_datum(operative_antag_datum_type)
+	time_limit = world.time + 45 MINUTES //Puts a hard cap on the time limit to avoid boredom.
+	addtimer(CALLBACK(src, .proc/check_win), 45.5 MINUTES)
+	SSstarsystem.add_blacklist(/obj/structure/overmap/syndicate/ai/carrier) //No. Just no. Please. God no.
+	SSstarsystem.add_blacklist(/obj/structure/overmap/syndicate/ai/patrol_cruiser) //Syndies only get LIGHT reinforcements.
+	return ..()
+
+/datum/game_mode/pvp/OnNukeExplosion(off_station)
+	..()
+	nukes_left--
+
+/datum/game_mode/pvp/check_win()
+	if (nukes_left == 0)
+		return TRUE
+	if(world.time >= time_limit)
```

Comment:
Why are you checking this here is there is a timer that calls this after 45.5 minuites?

Reply:
Failsafe check. Yes there is a timer that checks this, but the nuke may have gone off before then.